Kernel/Syscalls/thread.cpp

```diff
@@ -77,9 +77,11 @@ KResultOr<FlatPtr> Process::sys$create_thread(void* (*entry)(void*), Userspace<c
 
     PerformanceManager::add_thread_created_event(*thread);
 
-    SpinlockLocker lock(g_scheduler_lock);
-    thread->set_priority(requested_thread_priority);
-    thread->set_state(Thread::State::Runnable);
+    {
+        SpinlockLocker lock(thread->get_lock());
+        thread->set_priority(requested_thread_priority);
+        thread->set_state(Thread::State::Runnable);
+    }
     return thread->tid().value();
```

Comment:
This change doesn't seem necessary.

Reply:
Calling `set_priority` and `set_state` should require the lock to be held. I got started on a rebase to get it up to date but I'll have to finish it later tonight.

Reply:
Oh sorry, what I meant to say: I don't think we need the `{ }` scope here, right?
Or is there some point in releasing the lock before calling `tid()`?

Reply:
Oh I see. It is not strcitly necessary as access to `Thread::tid` doesn't need to be protected. But it shortens the time we hold the spinlock. The shorter we hold spinlocks, the better. Doesn't matter much on a single core system, but with multiple core. It's a minor optimization only.